pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1455,6 +1455,32 @@ $func$
 LANGUAGE PLPGSQL VOLATILE;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.delete_series_from_metric(series_id bigint, metric_name text)
+RETURNS BIGINT
+LANGUAGE PLPGSQL
+AS
+$$
+DECLARE
+    delete_stmt text;
+    num_chunks_touched bigint := 0;
+BEGIN
+    FOR delete_stmt in
+        SELECT FORMAT('DELETE FROM %1$I.%2$I WHERE series_id = %3$s', schema_name, table_name, series_id)
+        FROM (
+            SELECT (COALESCE(chc, ch)).* FROM pg_class c
+                INNER JOIN pg_namespace n ON c.relnamespace = n.oid
+                INNER JOIN _timescaledb_catalog.chunk ch ON (ch.schema_name, ch.table_name) = (n.nspname, c.relname)
+                LEFT JOIN _timescaledb_catalog.chunk chc ON ch.compressed_chunk_id = chc.id
+            WHERE c.oid IN (SELECT show_chunks(metric_name)::oid)
```

Comment:
@cevian Do we need to do anything different here with metric_name to make it safe? I don't really think so I think if we get one that isn't right we will just throw an error.

Only thing I don't know is whether we need to get a metric name that is the right length from another function or something or if that happens in Promscale somewhere before this happens?

Reply:
There's a function for that right?